Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -145,16 +150,20 @@ static std::unique_ptr<WPE::DRM::Connector> chooseConnector(int fd, drmModeRes*
 
 static std::unique_ptr<WPE::DRM::Crtc> chooseCrtcForConnector(int fd, drmModeRes* resources, const WPE::DRM::Connector& connector, GError** error)
 {
-    WPE::DRM::UniquePtr<drmModeEncoder> encoder(drmModeGetEncoder(fd, connector.encoderID()));
-    if (!encoder) {
-        g_set_error_literal(error, WPE_DISPLAY_ERROR, WPE_DISPLAY_ERROR_CONNECTION_FAILED, "Failed to get the connector encoder");
-        return nullptr;
-    }
+    /* Try the currently connected encoder+crtc */
+    for (int i = 0; i < resources->count_encoders; ++i) {
+
+        WPE::DRM::UniquePtr<drmModeEncoder> encoder(drmModeGetEncoder(fd, resources->encoders[i]));
+        if (!encoder)
+            continue;
 
-    for (int i = 0; i < resources->count_crtcs; ++i) {
-        if (resources->crtcs[i] == encoder->crtc_id) {
-            WPE::DRM::UniquePtr<drmModeCrtc> drmCrtc(drmModeGetCrtc(fd, resources->crtcs[i]));
-            return WPE::DRM::Crtc::create(fd, drmCrtc.get(), i);
+        for (int j = 0; j < resources->count_crtcs; ++j) {
```

Comment:
Maybe we could still use the same code (the one remove by this patch) but also use the possible_crtcs check. Like @ceyusa  does here https://coral.googlesource.com/imx-gst-plugins-bad/+/refs/tags/1.14.4+imx-6/sys/kms/gstkmssink.c#241

Reply:
The here point is that it can happen that the connector is not bound to an encoder.

In this scenario, the following check in `chooseCrtcForConnector()` is too restrictive because directly cuts the possibility of still finding a valid encoder for the connector:  
  
``` cpp
    if (!encoder) {
        g_set_error_literal(error, WPE_DISPLAY_ERROR, WPE_DISPLAY_ERROR_CONNECTION_FAILED, "Failed to get the connector encoder");
        return nullptr;
    }
```

This valid encoder it can by found by querying to the encoders if they can be a possible encoder for the connector:

``` cpp
    /* Try the currently connected encoder+crtc */
    for (int i = 0; i < resources->count_encoders; ++i) {

        WPE::DRM::UniquePtr<drmModeEncoder> encoder(drmModeGetEncoder(fd, resources->encoders[i]));
        if (!encoder)
            continue;

        for (int j = 0; j < resources->count_crtcs; ++j) {
            const uint32_t crtcMask = 1 << j;
            const uint32_t crtcId = resources->crtcs[j];
            if (encoder->possible_crtcs & crtcMask) {
                WPE::DRM::UniquePtr<drmModeCrtc> drmCrtc(drmModeGetCrtc(fd, crtcId));
                return WPE::DRM::Crtc::create(fd, drmCrtc.get(), j);
            }
        }
    }
```